src/docs.js

```diff
@@ -160,18 +162,32 @@ async function downloader(options) {
     project: null,
   });
 
-  let auth = new client.Auth({
-    credentials: options.credentials,
-  });
+  let readStream = {};
+
+  if (options.testing) {
 
-  let creds = await auth.awsS3Credentials('read-only', options.bucket, options.project + '/');
+    awsMock.config.basePath = path.join(__dirname, '..', 'test');
+    let s3 = awsMock.S3(options.credentials);
```

Comment:
We generally want to keep testing code *only* in files under `test/`.  So the mock should be set up (and removed) in `test/main_test.js`, not in `src/docs.js`.

You can do this by allowing an `options._s3` which you use instead of creating a new `AWS.S3` if it is set.  The `_` suggests this option is private and in this case only for use from tests.